backend/mr/src/main/java/io/featurehub/mr/resources/PersonResource.java

```diff
@@ -53,6 +58,38 @@ public RegistrationUrl createPerson(CreatePersonDetails createPersonDetails, Cre
     Person currentUser = authManager.from(securityContext);
 
     if (authManager.isAnyAdmin(currentUser.getId().getId())) {
+
+      if (createPersonDetails.getPersonType() == PersonType.SERVICEACCOUNT) {
+        if (createPersonDetails.getName() == null) {
+          throw new BadRequestException("Name is required parameter");
+        }
+
+        final CreatedServicePerson servicePerson = personApi.createServicePerson(createPersonDetails.getName(),
+          currentUser.getId().getId());
+
+        if (servicePerson == null) {
+          throw new BadRequestException();
+        }
+
+        if (createPersonDetails.getGroupIds() != null) {
+          createPersonDetails.getGroupIds().forEach(id -> {
+          //add user to the groups
+            groupApi.addPersonToGroup(id, servicePerson.getPerson().getId().getId(), Opts.empty());
```

Comment:
There are multiple references to `servicePerson.getPerson().getId().getId()` maybe better to move this to variable

Reply:
@rvowles ^^

Reply:
ok, i will review but if its inside this loop only then the JVM will optimise, ta